app/controllers/arrival/XmlNegativeAcknowledgementController.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.arrival
+
+import config.FrontendAppConfig
+import controllers.actions._
+import javax.inject.Inject
+import models.ArrivalId
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.libs.json.Json
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import renderer.Renderer
+import services.ArrivalMessageService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class XmlNegativeAcknowledgementController @Inject()(
+  override val messagesApi: MessagesApi,
+  identify: IdentifierAction,
+  cc: MessagesControllerComponents,
+  val frontendAppConfig: FrontendAppConfig,
+  arrivalMessageService: ArrivalMessageService,
+  renderer: Renderer
+)(implicit ec: ExecutionContext)
+    extends FrontendController(cc)
+    with I18nSupport {
+
+  def onPageLoad(arrivalId: ArrivalId): Action[AnyContent] = identify.async {
+    implicit request =>
+      arrivalMessageService.getXMLSubmissionNegativeAcknowledgementMessage(arrivalId).flatMap {
+        case Some(rejectionMessage) =>
+          val json = Json.obj("nctsEnquiries" -> frontendAppConfig.nctsEnquiriesUrl, "functionalError" -> rejectionMessage.error)
+
+          renderer.render("xmlNegativeAcknowledgement.njk", json).map(Ok(_))
+        case _ => Future.successful(Redirect(controllers.routes.TechnicalDifficultiesController.onPageLoad()))
```

Comment:
Fo this we should return a 500 and render the technical difficulties page instead

Reply:
@amaalali As it was decided in one of our meetings to create a new ticket to address the mentioned issue do we need to implement the changes as a part of this PR